### PR TITLE
[hsakmt_roct] Added musl support

### PR DESCRIPTION
**Pull request by jpsamaroo**

### CODE REVIEW

**Comment by jpsamaroo** [comment]
@Micket

**Review by Micket — APPROVED** [review]
lgtm. Bumpt to 3.8 while you're at it?

File: H/hsakmt_roct/build_tarballs.jl [hunk 1]
```diff
@@ -30,6 +34,7 @@ make install
 # ROCT-Thunk-Interface only supports Linux, seemingly only 64bit
 platforms = [
     Linux(:x86_64, libc=:glibc),
+    Linux(:x86_64, libc=:musl),
```
Comment (giordano) [inline H/hsakmt_roct/build_tarballs.jl:37]: In Yggdrasil we're now using Julia v1.6 which has new types for the platforms:
```suggestion
    Platform("x86_64", "linux"; libc=:glibc),
    Platform("x86_64", "linux"; libc=:musl),
```

File: H/hsakmt_roct/build_tarballs.jl [hunk 2]
```diff
@@ -29,7 +33,8 @@ make install
 # platforms are passed in on the command line
 # ROCT-Thunk-Interface only supports Linux, seemingly only 64bit
 platforms = [
-    Linux(:x86_64, libc=:glibc),
+    Platform("x86_64", "linux"; libc=:glibc),
+    Platform("x86_64", "linux"; libc=:musl),
```
Comment (giordano) [inline H/hsakmt_roct/build_tarballs.jl:37]: And I managed to get them wrong
```suggestion
    Platform("x86_64", "linux"; libc="glibc"),
    Platform("x86_64", "linux"; libc="musl"),
```

**Comment by giordano** [comment]
@jpsamaroo do you plan to bump to v3.8 or we can merge as it is?

**Comment by jpsamaroo** [comment]
@giordano bumped to 3.8.0 and ready to merge once CI passes

**Pull request merged**